packages/rocketchat-api/server/v1/users.js

```diff
@@ -369,10 +369,12 @@ RocketChat.API.v1.addRoute('users.setPreferences', { authRequired: true }, {
 });
 
 /**
-	This API returns the logged user roles.
+  DEPRECATED
+ // TODO: Remove this after three versions have been released. That means at 0.66 this should be gone.
+ This API returns the logged user roles.
 
-	Method: GET
-	Route: api/v1/user.roles
+ Method: GET
+ Route: api/v1/user.roles
  */
 RocketChat.API.v1.addRoute('user.roles', { authRequired: true }, {
```

Comment:
Please add a `console.warn('The endpoint /api/v1/user.roles is deprecated and will be removed after version v0.xx.')`. Also, add it to the resulting object but only when we are in a development environment. Which, might need to add a helper method which generates everything I just mentioned with providing the endpoint and the version it will be removed.

We will need to update the deprecation warnings for the real version after it has been merged.